### PR TITLE
fix(stringcache): use correct go-redis method to get string value

### PR DESCRIPTION
**Pull request by pitasi**
Apparently I was using the wrong method so instead of returning

```
https://s3.amazonaws.com/keybase_processed_uploads/a3c1904616e3b1d6177a0b3770af3405_360_360.jpg
```

it is returning the full redis command:

```
get api-server/validator-avatars/C5B260EC48CF27CD: https://s3.amazonaws.com/keybase_processed_uploads/a3c1904616e3b1d6177a0b3770af3405_360_360.jpg
```

sorry 😓  I did manual test my previous pr but probably didn't pay enough attention?

### CODE REVIEW

**Review by DeshErBojhaa — COMMENTED** [review]
Tests?

**Comment by pitasi** [comment]
> Tests?

I don't know how to test this. Do you know if we already have tests involving redis?

**Comment by DeshErBojhaa** [comment]
> > Tests?
> 
> I don't know how to test this. Do you know if we already have tests involving redis?

When we used redis in price-oracle, we used miniredis to have a temp redis cluster. You can try using that.

**Comment by pitasi** [comment]
@DeshErBojhaa miniredis is great! I was able to write some unit tests for the storebackend 😁

File: lib/stringcache/storebackend_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+func TestStoreBackend_SetNewKey(t *testing.T) {
+	assert := assert.New(t)
+	s, _ := NewStore(t)
+
+	back := NewStoreBackend(s)
+	err := back.Set(context.Background(), "key", "value", 10*time.Minute)
+	assert.NoError(err)
+}
```
Comment (DeshErBojhaa) [inline lib/stringcache/storebackend_test.go]: Please consider closing.
```go
func NewStore(t *testing.T) (*store.Store, *miniredis.Miniredis, func(){}) {
	s := miniredis.RunT(t)
	storeClient, err := store.NewClient(s.Addr())
	if err != nil {
		t.Fatalf("creating a new Store: %v", err)
	}
	return storeClient, s, func(){s.close()}
}
```

And in tests,
```go
func TestStoreBackend_SetNewKey(t *testing.T) {
	assert := assert.New(t)
	s, _, tearDown := NewStore(t)
    defer tearDown()

	back := NewStoreBackend(s)
	err := back.Set(context.Background(), "key", "value", 10*time.Minute)
	assert.NoError(err)
    s.FastForward(11*time.Minute)
    require().False(s.Exists("key"))
}
```

**Review by DeshErBojhaa — APPROVED** [review]
If we use require in other test files, please also use require here. It will keep things consistent across the repo.

Also please address the tdonn thing I commented.

After that consider this approved.

File: lib/stringcache/storebackend_test.go (hunk 1 above)
Comment (pitasi) [reply lib/stringcache/storebackend_test.go]: no need to explicit close as it's registered in the scope of the test (note the `t` param in `miniredis.RunT(t)`)

from their docs:
```
// RunT start a new miniredis, pass it a testing.T. It also registers the cleanup after your test is done.
```

Comment (pitasi) [reply lib/stringcache/storebackend_test.go]: btw what it does internally is just call

```
	t.Cleanup(m.Close)
```

which is pretty neat

**Comment by pitasi** [comment]
> If we use require in other test files, please also use require here. It will keep things consistent across the repo.

agree! done!

**Review by gsora — APPROVED** [review]

**Pull request merged**